x-pack/plugins/alerting/README.md

```diff
@@ -0,0 +1,232 @@
+
+**server.plugins.alerting.registerType(options)**
+
+The following table describes the properties of the `options` object.
+
+|Property|Description|Type|
+|---|---|---|
+|id|Unique identifier for the alert type.|string|
+|name|A user friendly name for the alert type.|string|
+|validate.params|Joi object validation for the parameters the executor receives.|Joi schema|
+|execute|A function to be called when executing an alert. See executor below.|Function|
+
+### Executor
+
+This is the primary function for an alert type, whenever the alert needs to execute, this function will perform the execution. It receives a variety of parameters, the following table describes the properties the executor receives.
+
+**execute(options)**
+
+|Property|Description|
+|---|---|
+|services.callCluster|Use this to do elasticsearch queries on the cluster Kibana connects to. NOTE: This currently authenticates as the Kibana internal user, this will change in a future PR.|
+|services.savedObjectsClient|Use this to manipulate saved objects. NOTE: This currently only works when security is disabled. A future PR will add support for enabled security using Elasticsearch API tokens.|
+|services.log|Use this to create server logs. (This is the same function as server.log)|
+|scheduledRunAt|The date and time the alert type was supposed to be called.|
+|previousScheduledRunAt|The previous date and time the alert type was supposed to be called.|
+|params|Parameters for the execution.|
+|state|State returned from previous execution.|
+
+### Example
+
+```
+server.plugins.alerting.registerType({
+  id: 'my-alert-type',
+  name: 'My alert type',
+  validate: {
+    params: Joi.object()
+      .keys({
+        myParam: Joi.string().required(),
+      })
+      .required(),
+  },
+  async execute({
+    scheduledRunAt,
+    previousScheduledRunAt,
+    services,
+    params,
+    state,
+  }: AlertExecuteOptions) {
+    // Pass in parameters, also validated
+    const {
+      myParam
+    } = params;
+
+    // Available services
+    const {
+      log,
+      callCluster,
+      savedObjectsClient,
+      alertInstanceFactory,
+    } = services;
+  
+    // Firing actions
+    alertInstanceFactory('server_1')
+      .replaceState({
+        // Alert instance level state, use getState() for
+        // previous and persisted values
+        ...
+      })
+      .fire('default', {
+        server: 'server_1',
+      });
+    
+    // Returning updated alert type level state
+    return {
+      ...
+    };
+  },
+});
+```
+
+## Alerts
+
+Using an alert type requires an alert to be created which will contain parameters and actions for a given alert type.
+
+The alerting plugin exposes the following APIs:
+
+#### `POST /api/alert`: Create alert
+
+Payload:
+
+|Property|Description|Type|
+|---|---|---|
+|alertTypeId|The id value of the alert type you want to call when the alert is scheduled to execute.|string|
+|interval|The interval in milliseconds the alert should execute.|number|
```

Comment:
thinking into the _near_ future, we'll want some `crontab` like thing here.  Here's how watcher does this: https://www.elastic.co/guide/en/elastic-stack-overview/7.1/trigger-schedule.html 

I propose we eventually follow that, we can do just [`interval`](https://www.elastic.co/guide/en/elastic-stack-overview/7.1/trigger-schedule.html#schedule-interval) for now, unitless, and should probably be seconds and not millseconds.

That just means moving `interval` under a new top-level `schedule` attribute.  We can then roll out the other schedule types as needed.